server/src/main/java/org/elasticsearch/transport/ProxyConnectionStrategy.java

```diff
@@ -167,7 +167,11 @@ public class ProxyConnectionStrategy extends RemoteConnectionStrategy {
         assert Strings.isEmpty(configuredAddress) == false : "Cannot use proxy connection strategy with no configured addresses";
         this.address = address;
         this.clusterNameValidator = (newConnection, actualProfile, listener) -> transportService.handshake(
-            RemoteConnectionManager.wrapConnectionWithClusterAlias(newConnection, clusterAlias),
+            RemoteConnectionManager.wrapConnectionWithRemoteClusterInfo(
+                newConnection,
+                clusterAlias,
+                connectionManager.getConnectionProfile().getTransportProfile()
```

Comment:
Based on my (limited) understanding, the `connectionManager.getConnectionProfile()` seems to be always the same as `actualProfile`. Is there a reason for using `connectionManager.getConnectionProfile()` and not `actualProfile.getTransportProfile()`? If they can never be different, should we maybe `assert` that they are equal?

Reply:
Yes they must be identical. Adding assertion is a good idea. I added one for asserting the transport profile being identical (that's the field that we actually care about).

The remote cluster related code does not encapsulate the ConnectionProfile well in that it is passed around and available in serveral places. But in fact the value is always what the underlying ConnectionManager uses. I found it confusing since there are now multiple sources of truth. I initially had `RemoteConnectionStrategy` to store the `ConnectionProfile` but that's also adding to the problem. So I decided to consistently getting it from `connectionManager`. It is my view that we should be able to refactor the code to do that across the remote cluster related code and remove any double exposure of the connectionProfile. But that's not this PR.

Reply:
I also made similar change to `SniffConnectionStrategy.getConnectionValidator` (4d5f9f3d3429d8c7de9c29da5ed705e7e02ed3df).

Reply:
Thanks for the explanation and adding assertions 👍